cns/restserver/restserver.go

```diff
@@ -58,7 +59,20 @@ type HTTPRestService struct {
 	state                    *httpRestServiceState
 	podsPendingIPAssignment  *bounded.TimedSet
 	sync.RWMutex
-	dncPartitionKey string
+	dncPartitionKey    string
+	EndpointState      map[string]*EndpointInfo // key : container id
+	EndpointStateStore store.KeyValueStore
+}
+
+type EndpointInfo struct {
+	PodName       string
+	PodNamespace  string
+	IfnameToIPMap map[string]*IPInfo // key : interface name, value : IPInfo
+}
+
+type IPInfo struct {
+	IPv4 *net.IPNet
```

Comment:
can this be array of addresses? e.g []net.IPNet

Reply:
also why this is a pointer instead of net.ipnet?

Reply:
Why would it be an array? Would an ifname have multiple IP addresses?

Reply:
not today..if in future if we support multiple v4 address in interface, then it can be extensible